docs/enhancements/infrastructure-monitoring-service-api/alarms.md

```diff
@@ -0,0 +1,770 @@
+As soon as a new AlarmDefinition row is added, we generate a new UUID and add a row to probable_cause table automatically (leveraging postgres trigger function) .
+When a user queries for all or a specific probableCause, we will simply join the tables to get the data.
+
+```sql
+SELECT
+    probable_causes.probable_cause_id,
+    ad.alarm_name,
+    ad.alarm_description
+FROM probable_causes
+         JOIN alarm_definitions ad
+              on ad.alarm_definition_id = probable_causes.alarm_definition_id
+WHERE probable_causes.probable_cause_id = 'f5ac4ac7-0ff3-40a2-b305-77313c28136a';
+```
+
+## K8s resources
+We will need few K8s resources that will be eventually applied by the Operator. 
+
+#### Alarm server
+This is essentially a typical CRUD app and we need the following
+
+- Deployment: It should have one initiContainer to create a DB, one initContainers that perform DB migration (create/alter table etc) 
+  and one main container which starts the main server. 
+  No HA, so set replica to 1. It should also contain all the ENV variables needed to talk to postgres deployment DB_HOST, DB_PORT, DB_NAME, DB_USER etc.
+  Suitable resources should be provided but not much memory and CPU is need for CRUD apps. 
+- Secrets: DB creds and configs should be read from postgres deployment. 
+- Service: Expose and balance using `ClusterIP` (though to start with we will set replica to 1)
+- Ingress: Expose service so that it can be called by users from outside the cluster
+
+
+#### Postgres
+This deployment can be leveraged by many microservices by creating their own Database. 
+
+- Deployment: One replica should be good enough for our case. 
+  Default username, password and dbname (note this is simply to allow postgres pod initialize) should be provided. 
+  Deployment should also mount to the right path `/var/lib/postgresql/data` (please check the latest docs).
+- PersistentVolumeClaim: At least 20G PVC should be used. Using test, we used about 60k rows DB which took about ~2Gb
+- Service: ClusterIP should be good as it's only used within the cluster. 
+- Secrets: default creds needed to spin postgres, with type: Opaque. This secret will then be ready by 
+   - POSTGRES_USER: `o-ran`
+   - POSTGRES_PASSWORD: `o-ran`
+   - POSTGRES_DB: `o-ran` # Note: this is simply there to be explicit.  
+                                   If not provided `POSTGRES_USER` is used to create default DB. But ultimately this DB not used as each service will create their own.
+- ConfigMap: For others to know the hostname and port
+   - POSTGRES_HOST: "postgres.o-ran-namespace.svc.cluster.local"
+   - POSTGRES_PORT: "5432"
+
+With the Secrets and ConfigMap applied, an app may connect to the DB with the following given the service (assuming initContainer of app already created DB `o-ran-infrastructure-monitoring-alarms`):
+```go
+connStr := "postgres://o-ran:o-ran@postgres.o-ran-namespace.svc.cluster.local:5432/o-ran-infrastructure-monitoring-alarms?sslmode=disable"
```

Comment:
We should try to enable TLS on all inter-pod communications.

Reply:
Yeah good point. I have deliberately left out the TLS stuff for now. Like alertmanager webhook also has a `tls_config` that we may configure as well? Added a note under "## Future Updates" to make sure we fix this after the init dev work